go/vt/sqlparser/tracked_buffer.go

```diff
@@ -95,13 +100,18 @@ func (buf *TrackedBuffer) Myprintf(format string, values ...interface{}) {
 			}
 		case 'v':
 			node := values[fieldnum].(SQLNode)
-			if buf.nodeFormatter == nil {
-				node.Format(buf)
-			} else {
-				buf.nodeFormatter(buf, node)
-			}
+			buf.formatter(node)
 		case 'a':
 			buf.WriteArg(values[fieldnum].(string))
+		case 'p':
```

Comment:
This part feels like it can lead to future bugs. Is there a way to make %v handle this correctly?
The main concern is this: someone may call Myprintf with %v without knowing that they should have called ExprPrintf with %p.